libevm/legacy/legacy_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package legacy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/libevm/core/vm"
+)
+
+func TestPrecompiledStatefulContract_Upgrade(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		envGas        uint64
+		input         []byte
+		cRet          []byte
+		cRemainingGas uint64
+		cErr          error
```

Comment:
```suggestion
		returnBuf          []byte
		remainingGas uint64
		precompileErr          error
```

`c` doesn't provide any information until reading the implementation. Alternatively, for consitency, `precompileRet` instead of `returnBuf`. I don't think the remaining gas needs to be qualified as it can only be from the precompile.